File: frontend/src/parent-template.ts
import { html, LitElement } from 'lit';
import 'child-template.js';

class ParentTemplate extends LitElement {
  render() {
    return html`
      <div>Parent Template</div>
      <child-template id="childTemplate"></child-template>
    `;
  }
}

customElements.define('parent-template', ParentTemplate);

File: src/main/java/com/example/application/component/ParentTemplate.java
package com.example.application.component;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.component.template.Id;

@Tag("parent-template")
@JsModule("./src/parent-template.ts")
public class ParentTemplate extends LitTemplate {

    public ParentTemplate() {

    }

    @Id("childTemplate")
    private ChildTemplate child;
}